From 1e311a937345e481eb8241e79de48cc2c5b665cf Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Thu, 18 Nov 2010 23:03:44 +0100
Subject: ASoC: ABE DSP: Disable IRQ and remove OPP set on the close

In order to avoid IRQ or DMA request at the end of audio activity
ABE IRQ visibility should be disabled. Also remove a redundant call
to set the OPP that is not needed at close.

Change-Id: I3b39587ee1db4b4bb8e0f3dd3d616319f2909c67
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Chris Kelly <c-kelly@ti.com>
---
 sound/soc/omap/omap-abe-dsp.c |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:14:24.182073278 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:14:41.963323364 +0530
@@ -1804,11 +1804,10 @@
 	abe->fe_id = dai->id;
 	dev_dbg(&rtd->dev, "%s ID %d\n", __func__, dai->id);
 
-	aess_set_opp_mode();
-
-	if (!--abe->active)
+	if (!--abe->active) {
+		abe_disable_irq();
 		abe_dsp_shutdown();
-
+	}
 	pm_runtime_put_sync(&pdev->dev);
 
 	mutex_unlock(&abe->mutex);
